cmd/kar-controllers/app/server.go

```diff
@@ -72,6 +72,11 @@ func Run(opt *options.ServerOption) error {
 		return err
 	}
 
+	err = listenReadinessProbe(opt)
```

Comment:
I don't think that code will ever be called until the call to `listenHealthProbe` returns on shutdown, as I think `http.ListenAndServe` is blocking.